mypy_django_plugin/main.py

```diff
@@ -303,11 +304,16 @@ def get_type_analyze_hook(self, fullname: str) -> Optional[Callable[[AnalyzeType
 
     def get_dynamic_class_hook(self, fullname: str) -> Optional[Callable[[DynamicClassDefContext], None]]:
         # Create a new manager class definition when a manager's '.from_queryset' classmethod is called
-        if fullname.endswith("from_queryset"):
+        if fullname.endswith(".from_queryset"):
             class_name, _, _ = fullname.rpartition(".")
             info = self._get_typeinfo_or_none(class_name)
             if info and info.has_base(fullnames.BASE_MANAGER_CLASS_FULLNAME):
                 return create_new_manager_class_from_from_queryset_method
+        elif fullname.endswith(".as_manager"):
+            class_name, _, _ = fullname.rpartition(".")
```

Comment:
```suggestion
        elif method_name == ".as_manager":
```
Slightly less duplication this way at least. Would also like to move the `_get_typeinfo_or_none` call out, but I guess there's a significant overhead of doing that for everything

Reply:
I don't know how much overhead it is. But since it is some, I just kept them inlined.

This is the work that'll happen:
https://github.com/python/mypy/blob/fc335cb16315964b923eb1927e3aad1516891c28/mypy/lookup.py#L12